### PR TITLE
Doc changes for TLS and loadBalancerIP changes

### PR DESCRIPTION
**Pull request by devloop0**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup

/kind documentation

> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:
Clarifies usage about the `agones.allocator.{disableMTLS,disableTLS,http.loadBalancerIP}` flags.

**Which issue(s) this PR fixes**:

Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 1]
```diff
@@ -218,6 +218,8 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.allocator.generateTLS`                      | Set to true to generate TLS certificates or false to provide certificates in `certs/allocator/*`| `true`                 |
 | `agones.allocator.tolerations`                      | Allocator [toleration][toleration] labels for pod assignment                                    | `[]`                   |
 | `agones.allocator.affinity`                         | Allocator [affinity][affinity] settings for pod assignment                                      | `{}`                   |
+| `agones.allocator.disableTLS`                       | Turns off all authentication for incoming connections to the allocator.                         | `false`                |
```
Comment (pooneh-m) [inline site/content/en/docs/Installation/Install Agones/helm.md:221]: for `disableTLS`  how about turn off the built-in encryption and authentication mechanism for agones-allocator service.

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -295,6 +297,13 @@ For most use cases the controller would have required a restart anyway (eg: cont
 You can use our script located at {{< ghlink href="install/helm/agones/certs/cert.sh" >}}cert.sh{{< /ghlink >}} to generate them.
 {{< /alert >}}
 
+## Fixed Allocator Load Balancer IP
+
+If you have reserved a static IP for your allocator, in the case that the Agones allocator is installed as a `LoadBalancer` service, you can pass the static IP you have reserved to Agones with the `agones.allocator.http.loadBalancerIP`. Assuming your Kubernetes provider supports this setting a `LoadBalancer` service's IP, this will do two things:
```
Comment (pooneh-m) [inline site/content/en/docs/Installation/Install Agones/helm.md:302]: Let's try to reword this to make it even clearer. One suggestion:

In order to reuse the existing load-balancer IP on upgrade or to install the agones-allocator service using a reserved IP, a user can specify the load-balancer IP as a helm configuration value for `agones.allocator.http.loadBalancerIP`. By setting the `loadBalancerIP` value:

1. The load-balancer is created with the specified IP, if supported by the cloud provider.
2. A self-signed server TLS certificate is generated for the IP, used by the `agones-allocator` service.

**Comment by pooneh-m** [comment]
@markmandel can you please take a look?

File: site/content/en/docs/Advanced/allocator-service.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
 
 ## Server TLS certificate
 
-Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using [cert-manager.io](https://cert-manager.io/) solution for cluster level certificate management.
+Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using [cert-manager.io](https://cert-manager.io/) solution for cluster level certificate management. Note that if you installed the Agones allocator as a `LoadBalancer` service [with a static IP]({{< relref "/docs/Installation/Install Agones/helm.md#fixed-allocator-load-balancer-ip" >}}) on installation, you do not need to do this step, since the Helm file will deal with generating a CA cert corresponding to the IP you passed in.
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply site/content/en/docs/Advanced/allocator-service.md:32]: Please refer to https://agones.dev/site/docs/contribute/#within-a-page and see other commits to see some examples on how to use it properly.

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 4]
```diff
@@ -295,6 +297,13 @@ For most use cases the controller would have required a restart anyway (eg: cont
 You can use our script located at {{< ghlink href="install/helm/agones/certs/cert.sh" >}}cert.sh{{< /ghlink >}} to generate them.
 {{< /alert >}}
 
+## Fixed Allocator Load Balancer IP
```
[the remark this answers is not in the document]
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:300]: Done!

File: site/content/en/docs/Advanced/allocator-service.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (devloop0) [reply site/content/en/docs/Advanced/allocator-service.md:32]: Thanks! I gated this (and the previous section describing the loadBalancerIP flag), with 1.9.0.

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 2 above)
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:302]: Reworded it as you suggested!

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 1 above)
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:221]: Changed!

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 5]
```diff
@@ -218,6 +218,8 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.allocator.generateTLS`                      | Set to true to generate TLS certificates or false to provide certificates in `certs/allocator/*`| `true`                 |
 | `agones.allocator.tolerations`                      | Allocator [toleration][toleration] labels for pod assignment                                    | `[]`                   |
 | `agones.allocator.affinity`                         | Allocator [affinity][affinity] settings for pod assignment                                      | `{}`                   |
+| `agones.allocator.disableTLS`                       | Turns off the built-in encryption and authentication mechanism for the agones-allocator service. | `false`                |
```
Comment (aLekSer) [inline site/content/en/docs/Installation/Install Agones/helm.md:221]: This should be moved to line 229:
```suggestion
```

Comment (aLekSer) [reply site/content/en/docs/Installation/Install Agones/helm.md:221]: Note about `New Configuration Features` table below.

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 6]
```diff
@@ -218,6 +218,8 @@ The following tables lists the configurable parameters of the Agones chart and t
...
+| `agones.allocator.disableMTLS`                      | Just turns off client cert authentication for incoming connections to the allocator.            | `false`                |
```
Comment (aLekSer) [inline site/content/en/docs/Installation/Install Agones/helm.md:222]: Could be simplified a bit, I cannot understand the difference based on the descriptions.
```suggestion
| `agones.allocator.disableMTLS`                      | Turns off client cert authentication for incoming connections to the allocator.            | `false`                |
```

**Review by aLekSer — CHANGES_REQUESTED** [review]
Could you please use `New Configuration Features` for the new `disableTLS` flag, and I assume that disableMTLS could be left as is. Or both could be moved to a section under publishVersion:
https://github.com/googleforgames/agones/blame/638a7fddecd23ec24881a738a59c45e0f8b0db7d/site/content/en/docs/Installation/Install%20Agones/helm.md#L227

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 6 above)
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:222]: I clarified the text a bit.
disableMTLS - no client cert authentication
disableTLS -. full plaintext (no client authentication, and no server cert encryption)

**Comment by devloop0** [comment]
I moved both disableMTLS and disableTLS to the 'New Configuration Changes' section.

**Review by aLekSer — APPROVED** [review]
Thanks for the fast fix. Someone is needed to verify wording before the final merge.

**Review by pooneh-m — COMMENTED** [review]
Thanks for the changes.

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 7]
```diff
@@ -283,7 +285,7 @@ RUNNING: agones-test
 ERROR: pods "agones-test" already exists
 Error: 1 test(s) failed
 ```
-That mean that you skiped `--cleanup` flag and you should either delete `agones-test` pod manually or run with the same test `helm test my-release --cleanup` two more times.
+That mean that you skipped `--cleanup` flag and you should either delete `agones-test` pod manually or run with the same test `helm test my-release --cleanup` two more times.
```
Comment (roberthbailey) [inline site/content/en/docs/Installation/Install Agones/helm.md:288]: nit: That means that you skipped the `--cleanup` flag ... delete the `agones-test` pod ...

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 8]
```diff
@@ -229,6 +229,8 @@ The following tables lists the configurable parameters of the Agones chart and t
 | Parameter                                           | Description                                                                                     | Default                |
 | --------------------------------------------------- | ----------------------------------------------------------------------------------------------- | ---------------------- |
 | `agones.allocator.http.loadBalancerIP`              | The [Load Balancer IP][loadBalancerIP] of the Agones allocator load balancer. Only works if the Kubernetes provider supports this option. | ""                     |
+| `agones.allocator.disableMTLS`                      | Turns off client cert authentication for incoming connections to the allocator.            | `false`                |
```
[the remark this answers is not in the document]
Comment (aLekSer) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: One of this option was present, not sure if the functionality was the same as current one.

[the remark this answers is not in the document]
Comment (pooneh-m) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: There is a breaking change for `disableMTLS` between 1.8 and 1.9. The helm configuration was not documented because we did not want anyone get a dependency to it, until we know exactly what are the requirements for GKE ingress. If it is moved to 1.8 documentation, a user may get a dependency and the behavior changes with the next version.

[the remark this answers is not in the document]
Comment (roberthbailey) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: Even though it wasn't documented, should we add a 1.9 release note that mentions the breaking change?

[the remark this answers is not in the document]
Comment (pooneh-m) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: I don't think we have to document it as a breaking change as it has not been documented at all and off by default. Only by reading the code the feature flag could be discovered. However, if there is a best practices on what to do in this situation, we should go with it. Otherwise, I would say it is unnecessary.

File: site/content/en/docs/Advanced/allocator-service.md [hunk 9]
```diff
@@ -31,6 +31,10 @@ agones-allocator            LoadBalancer   10.55.251.73    <b>34.82.195.204</b>
 
 Replace the default server TLS certificate with a certificate with CN and subjectAltName. There are multiple approaches to generate a certificate. Agones recommends using [cert-manager.io](https://cert-manager.io/) solution for cluster level certificate management.
 
+{{% feature publishVersion="1.9.0" %}}
+If the `agones-allocator` service is installed as a `LoadBalancer` [using a static IP]({{< relref "/docs/Installation/Install Agones/helm.md#reserved-allocator-load-balancer-ip" >}}), a valid self-signed server TLS certificate is generated using the IP provided. Otherwise, the server TLS certificate should be replaced.
```
[the remark this answers is not in the document]
Comment (devloop0) [reply site/content/en/docs/Advanced/allocator-service.md:35]: Made it the first thing in this section. Done!

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 10]
```diff
@@ -229,6 +229,8 @@ The following tables lists the configurable parameters of the Agones chart and t
...
+| `agones.allocator.disableTLS`                       | Turns off client cert authentication and encryption using the server cert for incoming connections to the allocator. | `false`                |
```
[the remark this answers is not in the document]
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:233]: Done!

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 7 above)
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:288]: Fixed!

File: site/content/en/docs/Installation/Install Agones/helm.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (devloop0) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: +1 to what Pooneh mentioned. disableMTLS was never documented earlier.

[the remark this answers is not in the document]
Comment (markmandel) [reply site/content/en/docs/Installation/Install Agones/helm.md:232]: Just confirming before merging this - it is good to merge?

**Review by pooneh-m — APPROVED** [review]
Well done! Thanks for your contribution.
